Tvl.VisualStudio.JustMyCodeToggle/JustMyCodeTogglePackage.cs

```diff
@@ -26,27 +29,19 @@ public JustMyCodeTogglePackage()
             _command = new OleMenuCommand(invokeHandler, changeHandler, beforeQueryStatus, id);
         }
 
-        public SVsServiceProvider ServiceProvider
-        {
-            get
-            {
-                return new VsServiceProviderWrapper(this);
-            }
-        }
-
         public EnvDTE.DTE ApplicationObject
         {
             get
             {
-                return ServiceProvider.GetService(typeof(EnvDTE._DTE)) as EnvDTE.DTE;
+                return GetService(typeof(EnvDTE._DTE)) as EnvDTE.DTE;
```

Comment:
It's either `GetService`, or JTF.Run on `GetServiceAsync` in BeforeQueryStatus and InvokeHandler.

Reply:
I believe it will be called on the UI thread, so GetService is fine.

Reply:
And if somehow it wasn't called on the UI thread, we'd still need to switch to the UI thread because DTE isn't free-threaded, correct? So pretty much no matter what, this needs to be GetService?

Reply:
I've been using DTE on non-UI threads for years and it generally doesn't care. If you're being pedantic you should assume it isn't free-threaded. 😉